docs/reference/triggers.md

```diff
@@ -0,0 +1,42 @@
+# Triggers
+
+This is a list of the current options that are available to trigger an OpenFaaS Funtion
+
+## CLI
+Trigger a function using the `faas-cli` by using the function name
+
+```
+echo "triggered" | faas-cli invoke figlet
+```
+
+## HTTP
+This is the default, and standard method for interacting with your Functions.
+
+The function URL follows the pattern of:
+```
+https://<gateway URL>:<port>/function/<function name>
+```
+
+> There is also the ability to execute a function asynchronously by replacing `/function/` with `/async-function/` before the function name
+
+## Kafka
+Connect your function(s) to Kafka topics
+
+More information in the Incubator repository: [openfaas-incubator/kafka-connector](https://github.com/openfaas-incubator/kafka-connector)
+
+## AWS SNS
+Trigger a function from AWS SNS Notifications and Subscriptions
+
+More information in the repository: [affix/OpenFaaS-SNS](https://github.com/affix/OpenFaaS-SNS)
+
+## RabbitMQ
+Invoke functions from RabbitMQ topics
+
+More information in the repository: [Templum/rabbitmq-connector](https://github.com/Templum/rabbitmq-connector)
+
+## Azure EventGrid - CloudEvents
```

Comment:
Heading: CloudEvents

Within the body state it is an example for Azure EventGrid, but that any CloudEvent is supported in the function when sent over HTTP.

Reply:
Move this above Rabbit